src/main/java/org/owasp/benchmark/score/parsers/HCLReader.java

```diff
@@ -0,0 +1,146 @@
+		tr.setToolVersion("1.0");
+		while (line != null) {
+			try {
+				line = reader.readLine();
+				if (line != null) {
+					if (line.contains("writeVulnerabilityToFile")) {
+						parseFindings(tr, line);
+					} else if (line.contains("Agent Version:")) {
+						String version = line.substring(line.indexOf("Version:") + 8);
+						tr.setToolVersion(version.trim());
+					} else if (line.contains("[checking URL:") && line.contains("BenchmarkTest00001")) {
+						firstLine = line;
+					} else if (line.contains("[checking URL:")) {
+						lastLine = line;
+					}
+				}
+			} catch (Exception ex) {
+				ex.printStackTrace();
+			}
+		}
+		reader.close();
+		tr.setTime(calculateTime(firstLine, lastLine));
+		return tr;
+	}
+
+
+    private void parseFindings(TestResults tr, String json) throws Exception {
+        TestCaseResult tcr = new TestCaseResult();
+        
+        try {
+			String splitJson = json.substring(json.indexOf("{"));
+			JSONObject obj = new JSONObject(splitJson);
+			JSONObject result = obj.getJSONArray("issue-group").getJSONObject(0);
+			
+	        String ruleId = result.getJSONObject("issue-type").getString( "ref" );
+	        tcr.setCWE(cweLookup(ruleId));
+	        tcr.setCategory(ruleId);
+	
+	        JSONObject request = result.getJSONArray("variant-group").getJSONObject(0).getJSONObject("request");
+            String uri = request.getString("uri" );
+        
+	        if ( uri.contains( "BenchmarkTest" ) ) {
+		        String testNumber = uri.substring( uri.lastIndexOf('/') + "BenchmarkTest".length() + 1 );
+	            tcr.setNumber(Integer.parseInt(testNumber));
+		        if (tcr.getCWE() != 0) {
+		            // System.out.println( tcr.getNumber() + "\t" + tcr.getCWE() + "\t" + tcr.getCategory() );
+		            tr.put(tcr);
+		        }
+	        }
+        } catch (Exception e) {
+            // System.err.println("> Parse error: " + json);
+            // e.printStackTrace();
+        }
+    }
+
+    
+	private static int cweLookup(String rule) {
+		switch (rule) {
+		case "SessionManagement.Cookies":
+			return 614; // insecure cookie use
+		case "Injection.SQL":
+			return 89; // sql injection
+		case "Injection.OS":
+			return 78; // command injection
+		case "Injection.LDAP":
+			return 90; // ldap injection
+		case "CrossSiteScripting.Reflected":
+			return 79; // xss
+		case "Injection.XPath":
+			return 643; // xpath injection
+		case "PathTraversal":
+			return 22; // path traversal
+		case "Cryptography.Mac":
+			return 328; // weak hash
+		case "Cryptography.PoorEntropy":
+			return 330; // weak random
+		case "Cryptography.Ciphers":
+			return 327; // weak encryption
+		case "Validation.Required":
+			return 501; // trust boundary
+		}
+		return 0;
+	}
+
+	private String calculateTime(String firstLine, String lastLine) {
+		try {
+			String start = firstLine.split(" ")[0];
+			String stop = lastLine.split(" ")[0];
+			SimpleDateFormat sdf = new SimpleDateFormat("HH:mm:ss.SSS");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.